integrated_channels/integrated_channel/transmitters/__init__.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+"""
+Package for generic data transmitters which send serialized data to integrated channels.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+
+class Transmitter(object):
+    """
+    Interface for transmitting data to an integrated channel.
+
+    The interface contains the following method(s):
+
+    transmit(payload)
+        payload - The ``Exporter`` object expected to implement an ``export`` method that returns serialized data.
+    """
+
+    def __init__(self, enterprise_configuration, client=None):
+        """
+        Prepares a configuration and a client to be used to transmit data to an integrated channel.
+
+        Arguments:
+            * enterprise_configuration - The configuration connecting an enterprise to an integrated channel.
+            * client - The REST API client that'll transmit serialized data.
+        """
+        self.enterprise_configuration = enterprise_configuration
+        self.client = client(enterprise_configuration) if client is not None else None
```

Comment:
@UmanShahzad `if client is not None else None` seems a bit redundant? Is this just to make it explicit that `client` might be `None`? In that case we could consider adding an inline comment that mentions this, and just do `self.client = client(enterprise_configuration)` here.

Reply:
Yeah it's just to make sure we get an actual client. If I leave `client=None` here, it may seem a bit weird to then do just `client(enterprise_configuration)` to a reader. If I change it to always require a `client` as an argument, that may actually require a lot of other changes (since, from above, I wanted to keep a consistent interface if possible, and changing those will also require changing a bunch of tests, just for this).

I'll change it to `... if client else None` to make it a little less awkward sounding, though.

Reply:
That sounds great, thanks for the details!